samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -152,14 +165,15 @@ public ApplicationStatus status(StreamApplication streamApp) {
   }
 
   /**
-   * Create the Coordination needed by the application runner.
-   * @return {@link CoordinationUtils}
-   * @throws Exception exception
+   * Create the {@link CoordinationUtils} needed by the application runner.
+   * @return an instance of {@link CoordinationUtils}
+   * @throws Exception exceptions when creating the instance, such as {@link ClassNotFoundException} and
+   * {@link InstantiationException}
    */
-  /* package private */ CoordinationUtils getCoordinationUtils() throws Exception {
+  /* package private */ CoordinationUtils createCoordinationUtils() throws Exception {
     ApplicationConfig appConfig = new ApplicationConfig(config);
     String clazz = appConfig.getCoordinationServiceFactoryClass();
-    if (clazz != null) {
+    if (StringUtils.isNotEmpty(clazz)) {
       CoordinationServiceFactory factory = ClassLoaderHelper.fromClassName(clazz);
```

Comment:
Not sure why this needs to be a shared/pluggable CoordinatorService. Looks like this is just some utils for working with ZK?